src/node_modules/app/machines/builder/form.options.js

```diff
@@ -194,22 +280,22 @@ function computeRequestInternal (
 		agg.aggs = subresults.query;
 		isRequiredSet = isRequiredSet && subresults.isRequiredSet;
 	}
-	isRequiredSet = isRequiredSet && completions.every(c => {
-		const params = extendedParams[c.name];
+	isRequiredSet = isRequiredSet && aggParamsInfo.every(c => {
+		const params = computedParams[c.name];
 		return c.name === 'field'
 			|| !c.required
-			|| !c.name.startsWith('__')
-			&& isNotNil(params)
-			|| c.name.startsWith('__')
-			&& isNotNil(params?.[params?.__type])
+			|| is_xor(c.type)
+			&& isNotNil(params?.[params?.__selection])
+			|| isNotNil(params)
+	});
+	aggParamsInfo.forEach(param => {
+		processRequestValue(
+			computedParams,
+			param.name,
+			computedParams[param.name],
+			param.type
+		);
```

Comment:
Modifying an array in-place is really not acceptable, this needs to be done functionally.